### PR TITLE
[Actions] Use new action version and fix GCP action

### PR DESCRIPTION
**Pull request by Michaelvll**
The old action version uses the deprecated node js version. We update the version for those actions in this PR.

This PR also fixes the GCP action, which fails likely due to GH update their action logic to isolate the path across different runs.

### CODE REVIEW

File: .github/workflows/update-gcp-catalog.yml [hunk 1]
```diff
@@ -38,18 +38,28 @@ jobs:
           curl -O https://dl.google.com/dl/cloudsdk/channels/rapid/downloads/google-cloud-cli-436.0.0-linux-x86_64.tar.gz
           tar -xf google-cloud-cli-436.0.0-linux-x86_64.tar.gz
           ./google-cloud-sdk/install.sh -q
-          source ~/google-cloud-sdk/path.bash.inc
```
Comment (Michaelvll) [inline .github/workflows/update-gcp-catalog.yml:41]: Moved this to the next run, as GH action no longer share the path across run

File: .github/workflows/update-gcp-catalog.yml [hunk 2]
```diff
@@ -38,18 +38,28 @@ jobs:
...
         run: |
+          echo "> Sourcing gcloud path ..."
+          source ~/google-cloud-sdk/path.bash.inc
+          echo "> Writing service account key to file ..."
           echo "$GCP_SERVICE_ACCOUNT_KEY" > $HOME/service_account_key.json
+          echo "> Exporting service account key ..."
           export GOOGLE_APPLICATION_CREDENTIALS=$HOME/service_account_key.json
-          gcloud auth activate-service-account --key-file $GOOGLE_APPLICATION_CREDENTIALS &> /dev/null
+          echo "> Activating service account ..."
+          gcloud auth activate-service-account --key-file $GOOGLE_APPLICATION_CREDENTIALS
```
Comment (cblmemo) [inline .github/workflows/update-gcp-catalog.yml:52]: Will this show our secret service account name to the github actions?

File: .github/workflows/update-gcp-catalog.yml [hunk 3]
```diff
@@ -38,18 +38,28 @@ jobs:
...
+          echo "> Configuring project ..."
           gcloud config set project $GCP_PROJECT_ID &> /dev/null
+          echo "> Fetching catalog schema version ..."
           version=$(python -c 'import sky; print(sky.clouds.service_catalog.constants.CATALOG_SCHEMA_VERSION)')
+          echo "> Creating catalog directory ..."
           mkdir -p catalogs/catalogs/$version
           cd catalogs/catalogs/$version
+          echo "> Listing gcloud auths ..."
+          gcloud auth list
```
Comment (cblmemo) [inline .github/workflows/update-gcp-catalog.yml:61]: same, should we keep this? potentially leaking our account information

**Review by cblmemo — APPROVED** [review]
Thanks @Michaelvll ! It mostly looks good to me. Left some discussions about printing out our account information

**Comment by Michaelvll** [comment]
Thanks for the review @cblmemo! Those are good catches! Just updated. Merging now.

**Pull request merged**